### PR TITLE
WT-10371 Implement statistics framework for both providers

### PR DESCRIPTION
**Pull request by jiechenbo**

### CODE REVIEW

File: ext/storage_sources/azure_store/azure_storage_source.cpp [hunk 1]
```diff
@@ -43,15 +43,26 @@
 
 struct azure_file_system;
 struct azure_file_handle;
+
+// Statistics to be collected for the Azure blob storage.
+struct azure_statistics {
+    uint64_t list_objects_count;  // Number of list objects requests
+    uint64_t put_object_count;    // Number of put object requests
+    uint64_t read_object_count;   // Number of get object requests
+    uint64_t object_exists_count; // Number of object exists requests
```
Comment (korteland) [inline ext/storage_sources/azure_store/azure_storage_source.cpp:52]: I think the comments here (and in the GCP equivalent) are a bit redundant - the naming is clear enough already.

File: ext/storage_sources/azure_store/azure_storage_source.cpp [hunk 2]
```diff
@@ -662,6 +677,20 @@ azure_file_size(WT_FILE_HANDLE *file_handle, WT_SESSION *session, wt_off_t *size
     return 0;
 }
 
+// Log collected statistics.
+static void
+azure_log_statistics(const azure_store &azure_storage)
+{
+    azure_storage.log->log_debug_message(
+      "GCP list objects count: " + std::to_string(azure_storage.statistics.list_objects_count));
```
Comment (korteland) [inline ext/storage_sources/azure_store/azure_storage_source.cpp:685]: These are using still using GCP naming

File: ext/storage_sources/azure_store/azure_storage_source.cpp [hunk 3]
```diff
@@ -685,6 +714,9 @@ wiredtiger_extension_init(WT_CONNECTION *connection, WT_CONFIG_ARG *config)
         return ret != 0 ? ret : EINVAL;
     }
 
+    // Set up statistics.
```
Comment (korteland) [inline ext/storage_sources/azure_store/azure_storage_source.cpp:717]: This comment is also redundant (same for GCP).

**Review by ajmorton — APPROVED** [review]
LGTM

File: ext/storage_sources/azure_store/azure_storage_source.cpp (hunk 1 above)
Comment (jiechenbo) [reply ext/storage_sources/azure_store/azure_storage_source.cpp]: done! also renamed the variables [fcb039f](https://github.com/wiredtiger/wiredtiger/pull/8932/commits/fcb039f1e2c4c0dbf1e68bb8dad0aff3bebc9f24)

File: ext/storage_sources/azure_store/azure_storage_source.cpp (hunk 2 above)
Comment (jiechenbo) [reply ext/storage_sources/azure_store/azure_storage_source.cpp]: Oops! thank you for catching it out [fcb039f](https://github.com/wiredtiger/wiredtiger/pull/8932/commits/fcb039f1e2c4c0dbf1e68bb8dad0aff3bebc9f24)

File: ext/storage_sources/azure_store/azure_storage_source.cpp (hunk 3 above)
Comment (jiechenbo) [reply ext/storage_sources/azure_store/azure_storage_source.cpp]: removed [fcb039f](https://github.com/wiredtiger/wiredtiger/pull/8932/commits/fcb039f1e2c4c0dbf1e68bb8dad0aff3bebc9f24)

**Review by korteland — APPROVED** [review]
LGTM!

**Comment by jiechenbo** [comment]
evergreen merge

**Pull request merged**